### PR TITLE
FEN-44

### PR DESCRIPTION
**Pull request by CarlosCodify**
Create model, controller and rspec of significative_event and contingency

### CODE REVIEW

File: app/controllers/api/v1/invoices_controller.rb [hunk 1]
```diff
@@ -64,8 +64,16 @@ def create
           @invoice.qr_content = qr_content(@invoice.company_nit, @invoice.cuf, @invoice.number, 1)
           @invoice.save
 
-          send_client_email
+          # TODO: here or after create?
+          @client = @company.clients.find_by(code: invoice_params[:client_code])
+          @xml = generate_xml(@invoice)
 
+          SendSiatJob.perform_later(@xml, @branch_office)
```
Comment (codify-dev) [inline app/controllers/api/v1/invoices_controller.rb:71]: Remove this logic from this PR. It's not part of the scope.

File: Gemfile [hunk 2]
```diff
@@ -45,6 +45,8 @@ gem 'savon', '~> 2.12.0'
 gem 'wicked_pdf'
 gem 'wkhtmltopdf-binary'
 
+gem 'rubyzip'
```
Comment (codify-dev) [inline Gemfile:48]: Remove this gem from this PR.

File: config/application.rb [hunk 3]
```diff
@@ -10,6 +10,8 @@
 
 module FenecInvoices
   class Application < Rails::Application
+    config.middleware.use Rack::Deflater
```
Comment (codify-dev) [inline config/application.rb:13]: What's it intended for?

File: app/controllers/api/v1/branch_offices_controller.rb [hunk 4]
```diff
@@ -43,6 +43,23 @@ def destroy
         @branch_office.destroy
       end
 
+      # TODO: refactor
+      def create_contingecy
```
Comment (codify-dev) [inline app/controllers/api/v1/branch_offices_controller.rb:47]: Move this to its own controller

File: config/routes.rb [hunk 5]
```diff
@@ -37,12 +37,17 @@
         post 'siat/payment_methods'
         post 'siat/legends'
         post 'siat/measurements'
+        post 'siat/significative_events'
```
Comment (codify-dev) [inline config/routes.rb:40]: significative_events should go at the root level, not inside of branch office. Like this:
api/v1/significative_events

File: config/routes.rb [hunk 6]
```diff
@@ -37,12 +37,17 @@
...
+        post 'create_contingecy'
+        post 'update_contingecy'
```
Comment (codify-dev) [inline config/routes.rb:42]: `create` and `update` actions should go like this:
POST v1/branch_offices/:id/contingencies
PUT v1/contingencies/:id

File: config/application.rb (hunk 3 above)
Comment (CarlosCodify) [reply config/application.rb:13]: what that line does is send everything compressed in gzip supposedly. l will remove, it was a test of the other task

**Review by codify-dev — APPROVED** [review]

**Pull request merged**